lib/src/solana_wallet.dart

```diff
@@ -59,4 +57,37 @@ class SolanaWallet {
       return HDKeyPair.fromSeed(seed, _getHDPath(account));
     }
   }
+
+  /// Returns a Future that resolves to the result of signing
+  /// [data] with the private key held internally by a given
+  /// instance
+  Future<crypto.Signature> sign(List<int> data) =>
+      _ed25519.sign(data, keyPair: _keyPair);
+
+  /// Sign a solana program message
+  Future<SignedTx> signMessage(Message message) async {
+    final List<int> serializedMessage = message.serialize();
+    final signature = Signature.from(await sign(serializedMessage));
+
+    return SignedTx(
+      signatures: CompactArray.fromList([signature]),
+      message: message,
+    );
+  }
+
+  /// Create a transfer of [lamports] from this wallet to
+  /// [destination] transaction and sign it
+  Future<SignedTx> buildAndSignTransferTx({
```

Comment:
Let's remove this method. build**And**SignTransferTx is a code smell – a method should mainly do one thing.

Using it in client will be pretty straightforward:

```dart
final message = Message.transfer(
  recentBlockhash: await solanaClient.getRecentBlockhash(),
  lamports: transferredAmount,
  source: sourceWallet.address,
  destination: targetWallet.address,
);
final SignedTx signedTx = await sourceWallet.signMessage(message);
```